websocketOutportDriver/websocketOutportDriverNodePart_test.go

```diff
@@ -0,0 +1,117 @@
+package websocketOutportDriver
+
+import (
+	"errors"
+	"testing"
+
+	coreMock "github.com/ElrondNetwork/elrond-go-core/core/mock"
+	"github.com/ElrondNetwork/elrond-go-core/data/indexer"
+	"github.com/ElrondNetwork/elrond-go-core/data/typeConverters/uint64ByteSlice"
+	"github.com/ElrondNetwork/elrond-go-core/marshal"
+	"github.com/ElrondNetwork/elrond-go-core/websocketOutportDriver/data"
+	"github.com/ElrondNetwork/elrond-go-core/websocketOutportDriver/mock"
+	"github.com/stretchr/testify/require"
+)
+
+func TestNewWebsocketOutportDriverNodePart(t *testing.T) {
+	t.Parallel()
+
+	t.Run("nil marshalizer", func(t *testing.T) {
+		args := getMockArgs()
+		args.Marshalizer = nil
+
+		o, err := NewWebsocketOutportDriverNodePart(args)
+		require.Nil(t, o)
+		require.Equal(t, ErrNilMarshalizer, err)
+	})
+
+	t.Run("nil logger", func(t *testing.T) {
+		args := getMockArgs()
+		args.Log = nil
+
+		o, err := NewWebsocketOutportDriverNodePart(args)
+		require.Nil(t, o)
+		require.Equal(t, ErrNilLogger, err)
+	})
+
+	t.Run("should work", func(t *testing.T) {
+		args := getMockArgs()
+
+		o, err := NewWebsocketOutportDriverNodePart(args)
+		require.NotNil(t, o)
+		require.NoError(t, err)
+	})
+}
+
+func TestWebsocketOutportDriverNodePart_SaveBlock_ErrWhileSendingOnRoute(t *testing.T) {
+	t.Skip("this test will run continuously so it should be run only when trying to reach that code area")
+
+	expectedErr := errors.New("cannot send on route")
+	defer func() {
+		r := recover()
+		require.Contains(t, r, expectedErr.Error())
+	}()
+
+	args := getMockArgs()
+	args.WebsocketSender = &mock.WebSocketSenderStub{
+		SendOnRouteCalled: func(_ data.WsSendArgs) error {
+			return expectedErr
+		},
+	}
+	o, err := NewWebsocketOutportDriverNodePart(args)
+	require.NoError(t, err)
+
+	o.SaveBlock(&indexer.ArgsSaveBlockData{})
+}
+
+func TestWebsocketOutportDriverNodePart_SaveBlock_ShouldWork(t *testing.T) {
+	t.Parallel()
+
+	defer func() {
+		r := recover()
+		require.Nil(t, r)
+	}()
+	args := getMockArgs()
+	o, err := NewWebsocketOutportDriverNodePart(args)
+	require.NoError(t, err)
+
+	o.SaveBlock(&indexer.ArgsSaveBlockData{})
```

Comment:
I think you should add a check for returned error here and everywhere else you do `o.SaveBlock`

Reply:
done